packages/vite/src/node/server/openBrowser.ts

```diff
@@ -39,6 +41,16 @@ export function openBrowser(
   return false
 }
 
+export async function resolveBrowserUrl(server: ViteDevServer): Promise<string> {
+  const options = server.config.server
+  const hostname = await resolveHostname(options.host)
+  const port = options.port || 3000
```

Comment:
The actual port used is not necessarily `options.port` (unless strict port is enabled).

But there's already an existing utility function that determines the current server URL with the correct port:

https://github.com/vitejs/vite/blob/e8b61bb3cac6c9c3ef3fa33e43440cc646473696/packages/vite/src/node/utils.ts#L807

Reply:
Super, thanks!